app.js

```diff
@@ -30,10 +30,33 @@ function start(){
     setInterval(tick,15000)
   })
 }
+function mergeProfileAndStats(stats, data){
+  if(data){
+    stats.credits = data.money;
+    stats.power = data.power;
+  }
+  return stats;
+}
+function addProfileData(api, stats){
+    return new Promise(resolve => api.me((err, data) => resolve(mergeProfileAndStats(stats, data))));
+}
+function mergeLeaderboardAndStats(stats, data){
+    if(data && data.length){
+        data = data[data.length-1];
+        stats.gclRank = data.rank+1;
+    }
+    return stats;
+}
+function addLeaderboardData(api, stats){
+    let leaderboardUrl = `https://screeps.com/api/leaderboard/find?mode=world&username=${api.user.username}`;
```

Comment:
leaderboardUrl should not include `https://screeps.com`, should just start with `/api`